sdks/fabric/interoperation-node-sdk/src/AssetManager.ts

```diff
@@ -666,11 +697,69 @@ const StartHTLCAssetLockListener = (
     assetId: string,
     recipientECert: string,
     lockerECert: string,
-    lockCallback: (c: Contract, d: string, t: string, i: string, r: string, l: string, v: string) => any,
+    lockCallback: (c: Contract, d: string, t: string, i: string, r: string, l: string, v: Hash, timeout: number) => any,
 ): void => {
     StartHTLCEventListener(contract, 'LockAsset', contractId, assetType, assetId, -1, recipientECert, lockerECert, lockCallback);
 }
 
+// Byzantine Swaps: Timed counterpart for timed AssetLockListener for reversion
+const assetLockExpirationCallback = (
+    contract: Contract, 
+    contractID: string, 
+    assetType: string, 
+    assetID: string, 
+    recipientECert: string, 
+    lockerECert: string, 
+    hash: Hash,
+    expiryTime: number
+): void => {
+    // Compare expiryTimeSec with currentTimeSec, which is number of milliseconds since epoch (L174)
+    const currTimeSecs = Math.floor(Date.now()/1000);
+
+    const reclaimCallback = async (contract: Contract, assetType: string, assetID: string, recipientECert: string, contractID: string) => {
+        // Check if asset hasn't been claimed yet. If true, do nothing. If false, either do the following cases:
+        const [islocked, isLockedQueryError] = await helpers.handlePromise(isAssetLockedInHTLCqueryUsingContractId(contract, contractID));
+        
+        if (islocked == false) {
+            // CASE #1: Check GetHTLCHashPreImageByContractId(contractId)
+            const [result, evaluateError] = await helpers.handlePromise(
+                contract.evaluateTransaction("GetHTLCHashPreImageByContractId", contractID),
+            );
+            
+            // CASE #2: If the function above returns ANY error, call reclaimAssetInHTLC
+            if (evaluateError) {
+                // If it succeeds, no change
+                const [reclaimResult, isReclaimableQueryError] = await helpers.handlePromise(reclaimAssetInHTLCusingContractId(contract, contractID));
+                if (isReclaimableQueryError) {
+                    reclaimAssetInHTLC(contract, assetType, assetID, recipientECert);             
```

Comment:
the following retry logic should go inside this if condition, instead of this line.

Reply:
Or you know what you can delete all this logic, from line 732 to 735.
the retry logic is good enough.